.github/workflows/cd.yaml

```diff
@@ -0,0 +1,54 @@
+name: CD
+
+on:
+    push:
+     branches:
+       -  main
+    
+jobs:
+  release-please:
+    name: Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    outputs:
+      tag_name: ${{ steps.release.outputs.tag_name }}
+      release_created: ${{ steps.release.outputs.release_created }}
+    steps:
+      - uses: google-github-actions/release-please-action@v4
+        id: release
+        with:
+          release-type: maven
+  
+  call-build-maven:
+    needs: release-please
+    name: build with Maven
+    uses: ./.github/workflows/maven_build.yaml
+  
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    needs: 
+      - release-please
+      - call-build-maven
+    steps:
+        - name: checkout code
+          uses: actions/checkout@v4
+
+        - name: Set up JDK 11
+          uses: actions/setup-java@v4
+          with:
+            java-version: '11'
+            distribution: 'corretto'
+            cache: 'maven'
+            
+        - name: Publish to GitHub Packages Apache Maven
+          if: ${{needs.release-please.outputs.release_created}}
```

Comment:
We should move this `if` to the job level if this job remains.

As it is, on non-release pushes, it's provisioning a runner, doing all the setup, and then terminating on this step and doing nothing.

Reply:
agree